### ISSUE TITLE
"NOT PARSED" and "Can't locate object method" errors

### ISSUE THREAD

**Issue by thorstenkampe** [glensc/nagios-plugin-check_raid#215]
**Output of `check_raid -d`:**

```
check_raid 4.0.10
Visit <https://github.com/glensc/nagios-plugin-check_raid#reporting-bugs> how to report bugs
Please include output of **ALL** commands in bugreport

DEBUG EXEC: /bin/sudo -n -A /sbin/dmsetup status --noflush at /usr/lib64/nagios/plugins/check_raid.pl line 503.
DEBUG EXEC: /bin/sudo -n -A /sbin/arcconf GETSTATUS 1 at /usr/lib64/nagios/plugins/check_raid.pl line 503.
DEBUG EXEC: /bin/sudo -n -A /sbin/arcconf GETCONFIG 1 AL at /usr/lib64/nagios/plugins/check_raid.pl line 503.
NOT PARSED: [Array Information] [Array Number 0] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 101.
NOT PARSED: [Array Information] [   Name                                     : A] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 102.
NOT PARSED: [Array Information] [   Status                                   : Ok] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 103.
NOT PARSED: [Array Information] [   Interface                                : SATA] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 104.
NOT PARSED: [Array Information] [   Total Size                               : 22892544 MB] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 105.
NOT PARSED: [Array Information] [   Unused Size                              : 0 MB] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 106.
NOT PARSED: [Array Information] [   Block Size                               : 512 Bytes] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 107.
NOT PARSED: [Array Information] [   Array Utilization                        : 100.00% Used, 0.00% Unused] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 108.
NOT PARSED: [Array Information] [   Type                                     : Data] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 109.
NOT PARSED: [Array Information] [   Transformation Status                    : Not Applicable] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 110.
NOT PARSED: [Array Information] [   Spare Rebuild Mode                       : Dedicated] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 111.
NOT PARSED: [Array Information] [   Logical 0                                : Optimal (5, Data, 15261706 MB) ARRAY5] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 115.
NOT PARSED: [Array Information] [   Device 0                                 : Present (11444224MB, SATA, HDD, Channel:0, Device:0)             ZTN0HZ8H] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 119.
NOT PARSED: [Array Information] [   Device 2                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:2) 7HKZYM8F            ] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 120.
NOT PARSED: [Array Information] [   Device 3                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:3) 2SG03E6F            ] at /usr/lib64/nagios/plugins/check_raid.pl line 927, <$fh> line 121.
Can't locate object method "process_array_information" via package "App::Monitoring::Plugin::CheckRaid::Plugins::arcconf" at /usr/lib64/nagios/plugins/check_raid.pl line 805, <$fh> line 125.
```

**Output of each command from `check_raid -d`**

/bin/sudo -n -A /sbin/dmsetup status --noflush
```
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--0d24bd92--eb59--47f7--95a7--5e20c0754aaf: 0 47292416 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--31dc0c15--7625--4a27--91bd--e960dcdb4ec4: 0 14753792 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--f8d84f02--0107--4aee--987b--2fab12106860: 0 10534912 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--5efae0cb--1064--4c8c--b74f--6cf49fd6061e: 0 214646784 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--6de8b045--428c--4353--944d--e0121bc14651: 0 4235264 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--3172dc5c--0ed1--4ebb--8266--f4f67e4c8873: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--a6ee6ecc--67fb--4a79--899d--08d861d19eca: 0 26337280 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--ad4cd57a--c408--4da0--b762--13e59a47634a: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--20d3eaa9--b12b--465c--9db3--6c53740b8100: 0 126091264 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-MGT: 0 8192 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--69f25cd5--027a--4695--ba72--11743534ba82: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--cbd770e2--b6f1--4c98--aca8--48843d2c03a7: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--7770f4d5--1a9e--417e--9f04--6879aa8340fb: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--a5c0c9fc--012b--45fb--b383--b9106b368d29: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--73ed88c9--3759--44f9--81d0--c4712d40b882: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--1f857070--8751--4638--a96b--51f3dfe3dbdd: 0 488538112 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--0c0799e3--f42d--475a--9c07--baf053159139: 0 47095808 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-spare: 0 1677721600 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--90f853e9--b753--46d5--94db--d360e91c6f8f: 0 35233792 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--d3545f06--4aa6--4504--bb58--521ab70e859c: 0 16384 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--eec2912c--372b--45d8--b4d1--7f1848e3fa1b: 0 840515584 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--8693f669--443c--4221--8721--245b52924ead: 0 246390784 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--6018ebb5--71c1--4bb5--9ac6--cf1f17cbf21a: 0 219725824 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--501282c2--a6e9--46d4--8685--eb0c28dcfa82: 0 32268288 linear 
VG_XenStorage--f2eea601--39b8--c430--cdec--0f43a10db8d2-VHD--c66e2aa0--8908--4d9f--82e8--ae8b776f705c: 0 16384 linear 
```

/bin/sudo -n -A /sbin/arcconf GETSTATUS 1
```
Controllers found: 1
   Current operation              : None

Command completed successfully.
```

/bin/sudo -n -A /sbin/arcconf GETCONFIG 1 AL
```
Controllers found: 1
----------------------------------------------------------------------
Controller information
----------------------------------------------------------------------
   Controller Status                        : Optimal
   Controller Mode                          : RAID (Hide RAW)
   Channel description                      : SCSI
   Controller Model                         : Adaptec Adaptec SmartRAID 3101-4
   Controller Serial Number                 : 9A37F301415
   Controller World Wide Name               : 50000D1E0024F780
   Physical Slot                            : 17
   Temperature                              : 62 C/ 143 F (Normal)
   Cache Serial Number                      : Not Applicable
   Cache memory                             : 784 MB
   Power Consumption                        : 7590 Watts 
   Power Mode                               : Maximum Performance
   Power Mode Operational                   : Maximum Performance
   Survival Mode                            : Enabled
   Host bus type                            : PCIe 3.0
   Host bus speed                           : 7880 MBps
   Host bus link width                      : 8 bit(s)/link(s)
   PCI Address (Bus:Device:Function)        : 0:2:0:0
   I2C Address                              : 0x0
   I2C Clock Speed                          : Disabled
   I2C Clock Stretching                     : Disabled
   Number of Ports                          : 1
   Internal Port Count                      : 1
   External Port Count                      : 0
   Defunct disk drive count                 : 0
   Logical devices/Failed/Degraded          : 1/0/0
   NCQ status                               : Enabled
   Queue Depth                              : Automatic
   Monitor and Performance Delay            : 60 minutes
   Elevator Sort                            : Enabled
   Degraded Mode Performance Optimization   : Disabled
   Latency                                  : Default
   Statistics data collection mode          : Disabled
   Global Physical Device Write Cache Policy: Disabled
   Cache Status                             : Not Configured
   Wait for Cache Room                      : Disabled
   Read Cache Percentage                    : 100 percent
   Write Cache Percentage                   : 0 percent
   No-Battery Write Cache                   : Disabled
   Write Cache Bypass Threshold Size        : 0 KB
   maxCache RAID5 WriteBack Enabled         : Disabled
   maxCache Version                         : Not Applicable
   Post Prompt Timeout                      : 0 seconds
   Boot Controller                          : False
   Primary Boot Volume                      : None
   Secondary Boot Volume                    : None
   Driver Name                              : smartpqi
   Driver Supports SSD Smart Path           : No
   Manufacturing Part Number                : 
   Manufacturing Spare Part Number          : 
   Manufacturing Wellness Log               : 
   --------------------------------------------------------
   RAID Properties
   --------------------------------------------------------
   Automatic Failover                       : Disabled
   Background consistency check             : Idle
   Consistency Check Delay                  : 3 seconds
   Parallel Consistency Check Supported     : Enabled
   Parallel Consistency Check Count         : 1
   Inconsistency Repair Policy              : Disabled
   Consistency Check Inconsistency Notify   : Disabled
   Rebuild Priority                         : High
   Expand Priority                          : Medium
   --------------------------------------------------------
   Controller Version Information
   --------------------------------------------------------
   Firmware                                 : 2.62[0]
   Driver                                   : Linux 1.2.1
   Hardware Revision                        : B
   --------------------------------------------------------
   Temperature Sensors Information
   --------------------------------------------------------
   Sensor ID                                : 0
   Current Value                            : 39 deg C
   Max Value Since Powered On               : 40 deg C
   Location                                 : Inlet Ambient

   Sensor ID                                : 1
   Current Value                            : 62 deg C
   Max Value Since Powered On               : 63 deg C
   Location                                 : ASIC

   Sensor ID                                : 2
   Current Value                            : 42 deg C
   Max Value Since Powered On               : 42 deg C
   Location                                 : Top

   Sensor ID                                : 3
   Current Value                            : 48 deg C
   Max Value Since Powered On               : 49 deg C
   Location                                 : Bottom


----------------------------------------------------------------------
Array Information
----------------------------------------------------------------------
Array Number 0
   Name                                     : A
   Status                                   : Ok
   Interface                                : SATA
   Total Size                               : 22892544 MB
   Unused Size                              : 0 MB
   Block Size                               : 512 Bytes
   Array Utilization                        : 100.00% Used, 0.00% Unused
   Type                                     : Data
   Transformation Status                    : Not Applicable
   Spare Rebuild Mode                       : Dedicated
   --------------------------------------------------------
   Array Logical Device Information
   --------------------------------------------------------
   Logical 0                                : Optimal (5, Data, 15261706 MB) ARRAY5
   --------------------------------------------------------
   Array Physical Device Information
   --------------------------------------------------------
   Device 0                                 : Present (11444224MB, SATA, HDD, Channel:0, Device:0)             ZTN0HZ8H
   Device 2                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:2) 7HKZYM8F            
   Device 3                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:3) 2SG03E6F            


----------------------------------------------------------------------
Logical device information
----------------------------------------------------------------------
Logical Device number 0
   Logical Device name                      : ARRAY5
   Disk Name                                : /dev/sda
   Array                                    : 0
   RAID level                               : 5
   Status of Logical Device                 : Optimal
   Size                                     : 15261706 MB
   Stripe-unit size                         : 256 KB
   Full Stripe Size                         : 512 KB
   Interface Type                           : Serial ATA
   Device Type                              : Data
   Boot Type                                : None
   Heads                                    : 255
   Sectors Per Track                        : 32
   Cylinders                                : 65535
   Caching                                  : Disabled
   Mount Points                             : /var/log 4096 MB  Partition Number 5 / 18432 MB  Partition Number 1 
   LD Acceleration Method                   : None
   Volume Unique Identifier                 : 600508B1001CB5AE64C0B1534D12EA8E
   --------------------------------------------------------
   Array Physical Device Information
   --------------------------------------------------------
   Device 0                                 : Present (11444224MB, SATA, HDD, Channel:0, Device:0)             ZTN0HZ8H
   Device 2                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:2) 7HKZYM8F            
   Device 3                                 : Present (7630885MB, SATA, HDD, Channel:0, Device:3) 2SG03E6F            


----------------------------------------------------------------------
Physical Device information
----------------------------------------------------------------------
   Channel #0:
      Device #0
         Device is a Hard drive
         State                              : Online
         Drive has stale RIS data           : False
         Block Size                         : 512 Bytes
         Physical Block Size                : 4K Bytes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,0(0:0)
         Reported Location                  : Enclosure Direct Attached, Slot 0(Connector 0:CN0)
         Array                              : 0
         Vendor                             : ATA     
         Model                              : ST12000NM003G-2M
         Firmware                           : SS03    
         Serial number                      : ZTN0HZ8H
         World-wide name                    : 30000D1E0024F780
         Reserved Size                      : 32768 KB
         Used Size                          : 11444192 MB
         Unused Size                        : 0 MB
         Total Size                         : 11444224 MB
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         SSD                                : No
         NCQ supported                      : Supported
         NCQ status                         : Enabled
         Boot Type                          : None
         Rotational Speed                   : 7200 RPM
         Current Temperature                : 30 deg C
         Maximum Temperature                : 51 deg C
         Threshold Temperature              : 60 deg C
         PHY Count                          : 1
         Drive Configuration Type           : Data
         Mount Point(s)                     : Not Mounted
         Drive Exposed to OS                : False
         Sanitize Erase Support             : True
         Drive Unique ID                    : 36906077088394709DC51BF7383C621D
         Last Failure Reason                : No Failure
      ----------------------------------------------------------------
      Device Phy Information                
      ----------------------------------------------------------------
         Phy #0
            Negotiated Link Rate            : 6 Gbps 
            Maximum Link Rate               : 6 Gbps 

      ----------------------------------------------------------------
      Device Error Counters                 
      ----------------------------------------------------------------
         Aborted Commands                   : 0
         Bad Target Errors                  : 0
         Ecc Recovered Read Errors          : 0
         Failed Read Recovers               : 0
         Failed Write Recovers              : 0
         Format Errors                      : 0
         Hardware Errors                    : 0
         Hard Read Errors                   : 0
         Hard Write Errors                  : 0
         Hot Plug Count                     : 0
         Media Failures                     : 0
         Not Ready Errors                   : 0
         Other Time Out Errors              : 0
         Predictive Failures                : 0
         Retry Recovered Read Errors        : 0
         Retry Recovered Write Errors       : 0
         Scsi Bus Faults                    : 0

      Device #2
         Device is a Hard drive
         State                              : Online
         Drive has stale RIS data           : False
         Block Size                         : 512 Bytes
         Physical Block Size                : 4K Bytes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,2(2:0)
         Reported Location                  : Enclosure Direct Attached, Slot 2(Connector 0:CN0)
         Array                              : 0
         Vendor                             : ATA     
         Model                              : HGST HUH721008AL
         Firmware                           : LHGNT9G0
         Serial number                      : 7HKZYM8F
         World-wide name                    : 30000D1E0024F781
         Reserved Size                      : 32768 KB
         Used Size                          : 7630853 MB
         Unused Size                        : 0 MB
         Total Size                         : 7630885 MB
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         SSD                                : No
         NCQ supported                      : Supported
         NCQ status                         : Enabled
         Boot Type                          : None
         Rotational Speed                   : 7200 RPM
         Current Temperature                : 30 deg C
         Maximum Temperature                : 36 deg C
         Threshold Temperature              : 60 deg C
         PHY Count                          : 1
         Drive Configuration Type           : Data
         Mount Point(s)                     : Not Mounted
         Drive Exposed to OS                : False
         Sanitize Erase Support             : True
         Drive Unique ID                    : 7F4A3E627CECB45616F119EC037900C0
         Last Failure Reason                : No Failure
      ----------------------------------------------------------------
      Device Phy Information                
      ----------------------------------------------------------------
         Phy #0
            Negotiated Link Rate            : 6 Gbps 
            Maximum Link Rate               : 6 Gbps 

      ----------------------------------------------------------------
      Device Error Counters                 
      ----------------------------------------------------------------
         Aborted Commands                   : 0
         Bad Target Errors                  : 0
         Ecc Recovered Read Errors          : 0
         Failed Read Recovers               : 0
         Failed Write Recovers              : 0
         Format Errors                      : 0
         Hardware Errors                    : 0
         Hard Read Errors                   : 0
         Hard Write Errors                  : 0
         Hot Plug Count                     : 0
         Media Failures                     : 0
         Not Ready Errors                   : 0
         Other Time Out Errors              : 0
         Predictive Failures                : 0
         Retry Recovered Read Errors        : 0
         Retry Recovered Write Errors       : 0
         Scsi Bus Faults                    : 0

      Device #3
         Device is a Hard drive
         State                              : Online
         Drive has stale RIS data           : False
         Block Size                         : 512 Bytes
         Physical Block Size                : 4K Bytes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,3(3:0)
         Reported Location                  : Enclosure Direct Attached, Slot 3(Connector 0:CN0)
         Array                              : 0
         Vendor                             : ATA     
         Model                              : HGST HUH721008AL
         Firmware                           : LHGNT9G0
         Serial number                      : 2SG03E6F
         World-wide name                    : 30000D1E0024F783
         Reserved Size                      : 32768 KB
         Used Size                          : 7630853 MB
         Unused Size                        : 0 MB
         Total Size                         : 7630885 MB
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         SSD                                : No
         NCQ supported                      : Supported
         NCQ status                         : Enabled
         Boot Type                          : None
         Rotational Speed                   : 7200 RPM
         Current Temperature                : 30 deg C
         Maximum Temperature                : 35 deg C
         Threshold Temperature              : 60 deg C
         PHY Count                          : 1
         Drive Configuration Type           : Data
         Mount Point(s)                     : Not Mounted
         Drive Exposed to OS                : False
         Sanitize Erase Support             : True
         Drive Unique ID                    : 95CB1207D3BB0989F67840E02F574453
         Last Failure Reason                : No Failure
      ----------------------------------------------------------------
      Device Phy Information                
      ----------------------------------------------------------------
         Phy #0
            Negotiated Link Rate            : 6 Gbps 
            Maximum Link Rate               : 6 Gbps 

      ----------------------------------------------------------------
      Device Error Counters                 
      ----------------------------------------------------------------
         Aborted Commands                   : 0
         Bad Target Errors                  : 0
         Ecc Recovered Read Errors          : 0
         Failed Read Recovers               : 0
         Failed Write Recovers              : 0
         Format Errors                      : 0
         Hardware Errors                    : 0
         Hard Read Errors                   : 0
         Hard Write Errors                  : 0
         Hot Plug Count                     : 0
         Media Failures                     : 0
         Not Ready Errors                   : 0
         Other Time Out Errors              : 0
         Predictive Failures                : 0
         Retry Recovered Read Errors        : 0
         Retry Recovered Write Errors       : 0
         Scsi Bus Faults                    : 0

   Channel #2:
      Device #1
         Device is an Enclosure Services Device
         Reported Channel,Device(T:L)       : 2,1(1:0)
         Enclosure Logical Identifier       : 50000D1E0024F784
         Type                               : SES2
         Vendor                             : Adaptec 
         Model                              : Virtual SGPIO
         Firmware                           : 2.62
         Status of Enclosure Services Device
            Speaker status                  : Not Available


----------------------------------------------------------------------
Connector information
----------------------------------------------------------------------
Connector #0
   Connector Name                           : CN0
   Functional Mode                          : RAID (Hide RAW)
   Connector Location                       : Internal
   SAS Address                              : 50000D1E0024F780



Command completed successfully.
```

**Additional environment details:**

Operating System: Citrix Hypervisor (XenServer) 8.2 CU1

**Comment by csware** [comment]
Not reproducible with latest master.

**Comment by csware** [comment]
Seems to be fixed with PR #210.
